blueprint/core/blueprint_ReactApplicationRoot.h

```diff
@@ -418,9 +418,6 @@ namespace blueprint
             JUCE_ASSERT_MESSAGE_THREAD
             jassert(handler);
 
-            // TODO: Here we have a case of `invoke` where the callback is not a temporary. Perhaps
-            // we want two different `invoke` methods: one for immeidately cleaning up temporaries,
-            // and one which persists native functions. `pinvoke`? `invokeWithPersistentNativeFunctions`?
             return engine.invoke("__BlueprintNative__.subscribe", eventType, std::move(handler));
```

Comment:
Ok, carrying on our convo from #89 here. As it is now, this commit will break this functionality because `engine.invoke` treats the callback as temporary, and deallocates it as soon as its slot in the internal temporary release pool is needed.

Based on the convo we started on the other PR, I guess the first question is probably: is this API even necessary? Because in one sense, (and I believe this is what you ended up doing @JoshMarler) if you want events from javascript, you can always just make your own `appRoot.registerNativeMethod("__MyApp__", "dispatchEventToNative"...)` and wire it up into your native code however you need.

I suppose I'd say that as we approach beta, it would help us to lean towards a minimal API, because if we announce a beta and users start working with blueprint more regularly, then introducing breaking API changes thereafter (e.g. removing something like approot.subscribe) becomes really hard. And introducing new API features is never a breaking change.

That said, we put this in originally for a reason, so obviously it has some value. A follow-up question then would be (if we decide to keep the API) how to implement it, as right now this would break

Reply:
Reading this on a phone so no further investigations but from experience my opinion is that this was poorly conceived request/change on my part. Initial the simplicity of this approach seemed appealing but I now feel it's API bloat. Specifying specific actions (be that a "dispatchEvent" function or something else) actually makes more sense to me and I think probably enforces more robust designs. Your native event/action handlers could take a variety of forms. If a completely generic 2-way data bind is what you need you can accomplish it very easily with the current functionality.

Reply:
Personally I think the sooner we pull it out the better. Maybe provide one clear foundational mechanism from native->JS communication and visa versa. Various patterns will begin to emerge in the wild no doubt and these can be documented as useful idioms as and when appropriate.

Reply:
Ok cool, yea, I'm with you. I think you're right that we'll see patterns emerge in the wild, and if any such pattern shows up that really starts to make sense as part of blueprint proper, we can always integrate it at that time. For now, we'll keep the API slim. I'll pull this out as part of this PR